WordPress/Classes/Stores/StatsInsightsStore.swift

```diff
@@ -778,7 +777,7 @@ private extension StatsInsightsStore {
 
     var hasCacheExpired: Bool {
         guard let siteID = SiteStatsInformation.sharedInstance.siteID,
-              let date = lastRefreshDate(for: siteID) else {
+              let date = lastRefreshDates[siteID] else {
```

Comment:
Since the cache is now stored in memory, this value also had to be moved to memory. It effectively solves the same problem of preventing the app from performing too many requests during its use.